nova_vm/src/ecmascript/builtins/primitive_objects.rs

```diff
@@ -183,7 +190,212 @@ impl InternalSlots for PrimitiveObject {
     }
 }
 
-impl InternalMethods for PrimitiveObject {}
+impl InternalMethods for PrimitiveObject {
+    fn internal_get_own_property(
+        self,
+        agent: &mut Agent,
+        property_key: PropertyKey,
+    ) -> JsResult<Option<PropertyDescriptor>> {
+        // For non-string primitive objects:
+        // 1. Return OrdinaryGetOwnProperty(O, P).
+        if let Some(backing_object) = self.get_backing_object(agent) {
```

Comment:
**issue:** String primitive objects can also have a backing store.

Reply:
Done.

Reply:
Ah, oh yeah I see. I guess it'd be a little more performant for strings to first check the index and length property descriptors first, as those are the most likely thing to be looked up. But! Primitive objects are stupid and rare and their performance doesn't matter at all, so here the deduplication in code that we achieve is definitely the better choice.

Reply:
Also, this is only suboptimal for string objects with backing objects, so that makes this deduplication even less worth the trouble.